src/fury/row/row.cc

```diff
@@ -237,8 +237,8 @@ int *ArrayData::GetDimensions(ArrayData &array, int num_dims) {
   // use deep-first search to search to numDimensions-1 layer to get dimensions.
   int depth = 0;
   auto dimensions = new int[num_dims];
-  int start_from_lefts[num_dims];
-  ArrayData *arrs[num_dims]; // root to current node
+  int *start_from_lefts = new int[num_dims];
+  ArrayData **arrs = new ArrayData *[num_dims]; // root to current node
```

Comment:
Why do you introduce two **memory-leaked allocation** here?

Reply:
`GetDimensions` can be removed, it's not used anymore